Updated the instructions about how to build a valid tx

ttmc opened the pull request:
We've changed how to build a valid transaction in version 2.0 of the transaction spec. Now we sign (fulfill) first and compute the transaction ID (i.e. hash the transaction) last.

@sbellem I'm not sure of some things. Please read the steps with a critical eye. In particular:

1. When constructing the thing-to-be signed (i.e. a transaction with no id yet, and with all fulfillments set to null), do you have a key-value pair of the form {"id": null} in there, or do you have no "id" key and value in there at all?
2. Before signing `utx_bytes` (i.e. the bytes of the unfulfilled transaction), do you first compute a hash of `utx_bytes` and sign _that_? I think you mentioned something like this. If the answer is yes, is it a SHA3-256 hash?

Commit:
Updated instructions to build a valid tx

ttmc commented:
@sbellem  Looking at your PR https://github.com/bigchaindb/bigchaindb/pull/1899 it seems the answer to my first question is that there _is_ an "id" key-value pair in the thing-to-be-signed, and that is also included in what gets hashed to compute the id. I will update this PR accordingly.

sbellem commented:
Regarding the `id` field I decided to stay consistent with the `fulfillment` field which we also set to `null` before fulfilling.

Regarding whether to hash the message (transaction) before fulfilling I have not modified this right now. I am not sure about what is best, and what difference it makes.

Commit:
The thing-to-be-signed includes an 'id':null key-value pair

ttmc commented:
@sbellem You're right, having the "id":null key-value pair is consistent with "fulfillment":null 

I just updated this pull request (on how to construct a valid transaction), after you approved it... I changed it so the thing-to-be-signed now includes the "id":null key-value pair. (That key-value pair is also in the thing-to-be-hashed later.) Can you take another look at this PR before I merge it?

sbellem commented on docs/transactions/source/transaction-components/transaction-id.rst:
```diff
@@ -23,11 +24,8 @@ Here are the steps to compute it.
       "metadata": metadata
     }
```
Should the `id` field be added?

ttmc replied:
Oh, good catch, I forgot about that other page! I'll go fix it now.

sbellem commented on docs/transactions/source/transaction-components/transaction-id.rst:
```diff
@@ -23,11 +24,8 @@ Here are the steps to compute it.
     }
 
-Note how ``d`` is missing the "id" key and value.
+Note how ``d`` is missing the ``"id"`` key and value.
```
Wouldn't it be better to write something like:

```rst
Note how the ``"id"`` of ``d`` is not yet set.
```

Commit:
The thing-to-be-hashed includes an 'id':null key-value pair
